projects/components/src/confirmation-modal/confirmation-modal.component.ts

```diff
@@ -0,0 +1,64 @@
+import { ChangeDetectionStrategy, Component, Inject } from '@angular/core';
+import { ButtonRole } from '../button/button';
+import { ModalOverlayConfig } from '../overlay/modal/modal';
+import { ModalRef } from '../overlay/modal/modal-ref';
+import { POPOVER_DATA } from '../popover/popover';
+
+@Component({
+  selector: 'lib-confirmation-modal',
+  styleUrls: ['./confirmation-modal.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="confirmation-modal">
+      <div class="description">
+        {{ this.descriptionText }}
+      </div>
+      <div class="controls">
+        <ht-button
+          label="Cancel"
+          class="cancel-button"
+          role="${ButtonRole.Tertiary}"
+          (click)="this.onCancel()"
+        ></ht-button>
+        <ht-button
+          [label]="this.actionButtonText"
+          class="confirm-button"
+          role="${ButtonRole.Additive}"
+          (click)="this.onConfirmation()"
+        ></ht-button>
+      </div>
+    </div>
+  `
+})
+export class ConfirmationModalComponent {
+  public actionButtonText?: string;
+  public descriptionText?: string;
+
+  public constructor(
+    private readonly modalRef: ModalRef,
+    @Inject(POPOVER_DATA) private readonly popoverData: ModalOverlayConfig<ConfirmationModalData>
```

Comment:
as far as a component using modals, I would expect it to be able to have two special injects available - `ModalRef` (can be a class-based inject, as it is here) and `MODAL_DATA` which is of an unknown type (so requires an inject token, similar to POPOVER_DATA), provided with the value of the data arg from the config.